rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -130,6 +134,13 @@ data LanguageService
     | implementer(Implementer implementer)
     ;
 
+@deprecated{ use builder and analyzer}
+LanguageService summarizer(Summarizer summarizer
+        , bool providesDocumentation = true
+        , bool providesDefinitions = true
+        , bool providesReferences = true
+        , bool providesImplementations = true) = builder(summarizer, providesDocumentation = providesDocumentation, providesDefinitions = providesDefinitions, providesReferences = providesReferences, providesImplementations = providesImplementations);
```

Comment:
what do you think @jurgenvinju ? I say it should map onto `analyzer` so that the builder is a new thing. and users might notice their analyzer running more often (but at least not, missing stuff).

Reply:
Analyzer should be the new thing and summarizers should become builders. The reason is that people would release their extension with a new version of our LSP bridge for a small change in their own code, and their users would accidentally get a huge performance regression as a "present". As a framework provide, we can't assume everybody does extensive testing every time. 

If we want that, then their code should not type-check anymore such that they are forced to choose which contribution kind is doing what they need. 

Since that would also create annoyance; I'm for the backwards compatible change.

Reply:
Fair point. I think we should print a message in the deprecation function to tell users to switch to either explicit builders. Or consider switching to using analyzers for on-demand type-checking.